File: csharp/Svix/Models/ListResponseEventTypeOut.cs
// this file is @generated
using System.Text.Json.Serialization;

namespace Svix.Models
{
    public class ListResponseEventTypeOut : BaseModel
    {
        [JsonPropertyName("data")]
        public required List<EventTypeOut> Data { get; set; }

        [JsonPropertyName("done")]
        public required bool Done { get; set; }

        [JsonPropertyName("iterator")]
        public required string? Iterator { get; set; }

        [JsonPropertyName("prevIterator")]
        public string? PrevIterator { get; set; }
    }
}

File: javascript/src/models/list_response_event_type_out.ts
// this file is @generated
import { EventTypeOut, EventTypeOutUtil } from "../models/event_type_out";

export interface ListResponseEventTypeOut {
  data: EventTypeOut[];
  done: boolean;
  iterator: string | null;
  prevIterator?: string | null;
}

export namespace ListResponseEventTypeOutUtil {
  export function _fromJsonObject(object: any): ListResponseEventTypeOut {
    return {
      data: object["data"].map((item) => EventTypeOutUtil._fromJsonObject(item)),
      done: object["done"],
      iterator: object["iterator"],
      prevIterator: object["prevIterator"],
    };
  }

  export function _toJsonObject(self: ListResponseEventTypeOut): any {
    return {
      data: self.data.map((item) => EventTypeOutUtil._toJsonObject(item)),
      done: self.done,
      iterator: self.iterator,
      prevIterator: self.prevIterator,
    };
  }
}
